### PR TITLE
Profile

### LINKED ISSUE #49: Profile Page UI
## What/Why?

As a user I want to have a profile page so that I can customize my account and manage my settings.

## Tasks

- [ ] Finalize figma design
- [ ] Create horizontal friends list widget
- [ ] Create profile page UI

### PR DESCRIPTION
**Pull request by echadwick6**
## What is this

Profile page skeleton. No backend included right now

## Reviewer

@mjbernet3 

## Setup

- [x] Ensure profile page pops up
- [x] Check if log out button works
- [x] Check if account and about pages work

## Steps to Test

- [x] Run the code
- [x] Click "log out"
- [x] Click "About" page
- [x] Click "Account" page

## Associated issues

closes #49

### CODE REVIEW

File: lib/frontend/components/discover/event_card.dart [hunk 1]
```diff
@@ -22,64 +23,72 @@ class SmallEventCard extends StatelessWidget {
...
-    return ClipRRect(
-      borderRadius: BorderRadius.all(
-        Radius.circular(10.0),
-      ),
-      child: DecoratedBox(
-        decoration: BoxDecoration(
-          color: AppColor.dp3,
+    return GestureDetector(
+      onTap: () {
+        Navigator.push(context, MaterialPageRoute(
```
Comment (mjbernet3) [inline lib/frontend/components/discover/event_card.dart]: Use named routes when pushing new pages to remain consistent with the rest of the application.

File: lib/frontend/components/profile/settings_list.dart [hunk 2]
```diff
@@ -0,0 +1,147 @@
+import 'package:flutter/material.dart';
+import 'package:project_fomo/backend/services/auth_service.dart';
+import 'package:provider/provider.dart';
+import 'package:project_fomo/style.dart';
+import 'package:custom_switch/custom_switch.dart';
+import 'package:project_fomo/frontend/pages/about_page.dart';
+import 'package:project_fomo/frontend/pages/account_page.dart';
+
+class SettingList extends StatelessWidget {
+
```
Comment (mjbernet3) [inline lib/frontend/components/profile/settings_list.dart]: These should be static constants.  Ex: static const TextStyle listName = ...

File: lib/frontend/components/profile/settings_list.dart [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+      decoration: BoxDecoration(
+        border: Border(
+          bottom: BorderSide(color: Color(0xFF343434)),
+        ),
+      ),
+  );
+
+  @override
+  Widget build(BuildContext context) {
+    AuthService authService = Provider.of<AuthService>(context);
```
Comment (mjbernet3) [inline lib/frontend/components/profile/settings_list.dart]: This is a note for myself, need to remove auth service from this widget.

File: lib/frontend/components/profile/settings_list.dart [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            Padding(
+              padding: EdgeInsets.only(top: 15, bottom: 15,),
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                  children: <Widget>[
+                    Text(
+                      "Location Services",
+                      style: listName,
+                    ),
+                    CustomSwitch(
```
Comment (mjbernet3) [inline lib/frontend/components/profile/settings_list.dart]: This is not complex enough to import a dependency and the package itself is not reliable enough.

File: lib/frontend/components/shared/gradient_icon.dart [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+class GradientIcon extends StatelessWidget {
+
+  final IconData icon;
+  final double size;
+
+  GradientIcon({@required this.icon, @required this.size});
+
+  @override
+  Widget build(BuildContext context) {
+    return ShaderMask(
```
Comment (mjbernet3) [inline lib/frontend/components/shared/gradient_icon.dart]: Are there performance considerations for shader mask?

File: lib/frontend/pages/event_page.dart [hunk 6]
```diff
@@ -1,31 +1,52 @@
...
+    imageUrl: 'https://pisces.bbystatic.com/image2/BestBuy_US/images/products/6358/6358944_sd.jpg;maxHeight=640;maxWidth=550');
+
 class EventPage extends StatelessWidget {
+
   static const String pageRoute = '/event-page';
+  final Event event;
+
+  EventPage({@required this.event});
+
+//  checkEvent() async {
```
Comment (mjbernet3) [inline lib/frontend/pages/event_page.dart]: Remove this commented code.

File: pubspec.yaml [hunk 7]
```diff
@@ -15,6 +15,7 @@ dependencies:
   carousel_slider: ^1.3.0
   provider: ^4.0.2
   cupertino_icons: ^0.1.2
+  custom_switch: ^0.0.1
```
Comment (mjbernet3) [inline pubspec.yaml]: Remove this dependency.

File: lib/frontend/components/shared/gradient_icon.dart (hunk 5 above)
Comment (mjbernet3) [inline lib/frontend/components/shared/gradient_icon.dart]: Can the existing gradient in the stylesheet not be used here?

**Comment by mjbernet3** [comment]
Need to always merge develop into your branch before creating a pull request.  Otherwise there will be tons of merge conflicts I do not have time to deal with them.

See comments above for other stuff.  I have not ran the code or tested any of it yet.

File: lib/frontend/components/profile/settings_list.dart [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+import 'package:project_fomo/backend/services/auth_service.dart';
+import 'package:provider/provider.dart';
+import 'package:project_fomo/style.dart';
+import 'package:custom_switch/custom_switch.dart';
+import 'package:project_fomo/frontend/pages/about_page.dart';
+import 'package:project_fomo/frontend/pages/account_page.dart';
+
+class SettingList extends StatelessWidget {
+
+  TextStyle listName = TextStyle(
```
Comment (mjbernet3) [inline lib/frontend/components/profile/settings_list.dart]: These should be static constants. Ex: static const TextStyle listName = ...

File: lib/frontend/components/shared/gradient_icon.dart (hunk 5 above)
Comment (echadwick6) [reply lib/frontend/components/shared/gradient_icon.dart]: It doesn't have the same effect. Unfortunately, that type of Gradient is just pure purple

File: lib/frontend/components/profile/settings_list.dart (hunk 4 above)
Comment (echadwick6) [reply lib/frontend/components/profile/settings_list.dart]: I'll implement backend stuff for pages at the end of spring break

**Pull request merged**